librad/src/net/peer.rs

```diff
@@ -194,10 +195,14 @@ where
         &self,
         urn: RadUrn,
     ) -> impl Future<Output = impl futures::Stream<Item = PeerInfo<IpAddr>>> {
+        let span = tracing::trace_span!("PeerApi::providers", urn = %urn);
         let protocol = self.protocol.clone();
-        let urn2 = urn.clone();
+        let target_urn = urn.clone();
+
         async move {
-            let events = protocol.subscribe().await.filter_map(move |evt| {
+            // Listen for `Has` gossip events where a provider
+            // claims to have the target urn.
+            let providers_stream = protocol.subscribe().await.filter_map(move |evt| {
```

Comment:
I'd prefer not to introduce hungarian notation and program flow comments. If it's unclear what the type of an expression is, or what it does, refactor it so you can add a type annotation and rustdoc commentary.

Reply:
I'm fine with dropping the comment, triggered by having `providers` doing two things.

Regarding the "Hungarian notation", I believe you are referring to the change from `let events` to `let providers_stream`? If so, I don't get your point, as "events" provides no clarity or meaning whereas "providers_stream" names it after exactly what it is.

Reply:
"Stream" is the type of the expression. `providers` would suffice as a name.